tools/gromacs/rmsd.xml

```diff
@@ -0,0 +1,150 @@
+                <param name="index_refgroup" value="0" min="0" type="integer" label="Index - least squares fitting group" help="Index number of the group you are fitting against, i.e. the reference group. For proteins, this group can be Backbone or C-alpha."/>
+            </when>
+            <when value="none">
+            </when>
+        </conditional>
+        <param name="outputformat" type="select" label="Output format">
+            <option value="xmgrace">Xmgrace. This is the default way.</option>
+            <option value="none">Raw Data.</option>
+        </param>
+        <param name="mir" type="select" label="Compare results with mirror image of the reference structure?" help="This is useful as a reference for assesing ‘significant’ RMSD values.">
+            <option value="false">No</option>
+            <option value="true">Yes</option>
+        </param>
+        <expand macro="log"/>
+    </inputs>
+    <outputs>
+        <data name="output1" format="xvg" from_work_dir="rmsd.xvg" label="GROMACS calculation of RMSD on ${on_string}"/>
+        <data name="output2" format="xvg" from_work_dir="rmsdmir.xvg" label="GROMACS calculation of mirror image RMSD on ${on_string}">    
+            <filter>mir == 'true'</filter>
+        </data>
+        <expand macro="log_outputs"/>
+    </outputs>
+    <tests>
+        <test>
+            <param name="traj_input" value="npt.xtc"/>
+            <param name="structure_input" value="npt.tpr" ftype="tpr"/>
+            <param name="ndx_input" value="index.ndx" ftype="ndx"/>
+            <param name="index" value="3"/>
+            <param name="outputformat" value="xmgrace"/>
+            <output name="output1" ftype="xvg">
+                <assert_contents>
+                    <has_text text="0.1000000    0.0391476"/>
+                    <has_text text="0.2000000    0.0598719"/>
+                    <has_text text="0.3000000    0.0842008"/>
+                </assert_contents>
+            </output>
+        </test>
+        <test>
+            <param name="traj_input" value="npt.xtc"/>
+            <param name="structure_input" value="npt.tpr" ftype="tpr"/>
+            <param name="ndx_input" value="index.ndx" ftype="ndx"/>
+            <param name="index" value="1"/>
+            <param name="resavg" value="true"/>
+            <param name="outputformat" value="none"/>
+            <output name="output1" ftype="xvg">
+                <assert_contents>
+                    <has_text text="0.1000000    0.0391476"/>
+                    <has_text text="0.2000000    0.0598719"/>
+                    <has_text text="0.3000000    0.0842008"/>
+                </assert_contents>
+            </output>
+        </test>
+    </tests>
+    <help><![CDATA[
+
+        .. class:: infomark
+        
+        **What it does**
+        
+        This tool calculates a structure's Root Mean Square Deviation (RMSD), using the built-in gromacs function gmx rms.
+        
+_____
```

Comment:
could you expand the help a bit more? I never used the rms subcommand but the help page has a lot of interesting information some of which could be useful. (I understand you haven't wrapped all functionality)

https://manual.gromacs.org/documentation/current/onlinehelp/gmx-rms.html

e.g. mention that the comparison is to the ref structure, that seems important. What if you use a second trajectory, is the structure not needed?

Reply:
Ok I will try to expand here. both trajectories use the same ref structure, as per my understanding.